ui/src/main/java/com/xmartlabs/bigbang/ui/common/recyclerview/BaseRecyclerViewAdapter.kt

```diff
@@ -244,9 +250,36 @@ abstract class BaseRecyclerViewAdapter : RecyclerView.Adapter<RecyclerView.ViewH
     addItems(type, newItems)
   }
   
+  /**
+   * Sets the items data for the recycler view and notifies any registered observers that the data set has
+   * changed.
+   * Unlike [setItems], this method updates the items and notifies removed/updated/changed items instead of a full
+   * reload.
+   *
+   * @param type     Type of items.
+   * *
+   * @param newItems The items tobe added.
+   */
+  @MainThread
+  fun <T : RecycleItemType<*, *>> updateItems(type: T, newItems: List<Any>) {
```

Comment:
is it needed? shouldn't you call the `setItems` with the diff function?

Reply:
Remove whole update items